### DISCUSSION TITLE
How to change stream track?

### DISCUSSION THREAD

**Discussion by JoanIzquierdo** [tweepy/tweepy#1339] [Questions] [answered]
I would like to change the stream track dynamically without losing any message. What's the best method for changing the track of a stream?

**Comment by MichaelCurrin** [comment]
You must stop and restart the stream. If you want to do this in a  script, using `stream.running = False` then start another filter stream.

https://michaelcurrin.github.io/python-twitter-guide/#/code_snippets?id=update-stream

**Comment by Harmon758** [comment, marked as answer]
Twitter's API does not include a way to change tracks dynamically for the same standard stream:
> One filter rule on one allowed connection, disconnection required to adjust rule

https://developer.twitter.com/en/docs/tweets/filter-realtime/overview

As of right now, you'll have to use the enterprise [PowerTrack API](https://developer.twitter.com/en/docs/tweets/filter-realtime/overview/powertrack-api) if you want to use a non-Labs endpoint to do so.

Otherwise, you'll have to create a new stream with the new track, as explained.
I'd recommend using `stream.disconnect` rather than manually modifying `stream.running` though.
See also the Missing data from disconnections section of [the documentation for consuming streaming data](https://developer.twitter.com/en/docs/tutorials/consuming-streaming-data).
